sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobBatch.java

```diff
@@ -0,0 +1,413 @@
+
+        this.accountUrl = accountUrl;
+
+        boolean batchHeadersPolicySet = false;
+        List<HttpPipelinePolicy> policies = new ArrayList<>();
+        for (int i = 0; i < pipeline.getPolicyCount(); i++) {
+            HttpPipelinePolicy policy = pipeline.getPolicy(i);
+
+            // Don't want these policies in the dummy pipeline.
+            if (policy instanceof HttpLoggingPolicy
+                || policy instanceof RetryPolicy
+                || policy instanceof RequestRetryPolicy
+                || policy instanceof UserAgentPolicy) {
+                continue;
+            }
+
+            if (policy instanceof SharedKeyCredentialPolicy) {
+                batchHeadersPolicySet = true;
+                // The batch policy needs to be added before the SharedKey policy to run preparation cleanup.
+                policies.add(new BatchHeadersPolicy());
+            }
+
+            policies.add(pipeline.getPolicy(i));
+        }
+
+        if (!batchHeadersPolicySet) {
+            policies.add(new BatchHeadersPolicy());
+        }
+
+        this.batchPipeline = new HttpPipelineBuilder()
+            .policies(policies.toArray(new HttpPipelinePolicy[0]))
+            .httpClient(new BatchClient(this::sendCallback))
+            .build();
+
+        this.batchOperationQueue = new ConcurrentLinkedDeque<>();
+        this.batchRequest = new ArrayList<>();
+        this.batchMapping = new ConcurrentHashMap<>();
+    }
+
+    /**
+     * Adds a delete blob operation to the batch.
+     *
+     * @param containerName The container of the blob.
+     * @param blobName The name of the blob.
+     * @return a {@link Response} that will be used to associate this operation to the response when the batch is
+     * submitted.
+     * @throws UnsupportedOperationException If this batch has already added an operation of another type.
+     */
+    public Response<Void> delete(String containerName, String blobName) {
+        return deleteHelper(buildClient(containerName, blobName), null, null);
+    }
+
+    /**
+     * Adds a delete blob operation to the batch.
+     *
+     * @param containerName The container of the blob.
+     * @param blobName The name of the blob.
+     * @param deleteOptions Delete options for the blob and its snapshots.
+     * @param blobAccessConditions Additional access conditions that must be met to allow this operation.
+     * @return a {@link Response} that will be used to associate this operation to the response when the batch is
+     * submitted.
+     * @throws UnsupportedOperationException If this batch has already added an operation of another type.
+     */
+    public Response<Void> delete(String containerName, String blobName,
+        DeleteSnapshotsOptionType deleteOptions, BlobAccessConditions blobAccessConditions) {
+        return deleteHelper(buildClient(containerName, blobName), deleteOptions, blobAccessConditions);
```

Comment:
Why is it necessary to build a client for each operation?

Reply:
I think because we need each request to go through a pipeline so it can get the data and auth stuff added, and for that we need a client per operation since each client can only point to one resource

Reply:
That is correct, each request needs to be funneled through a pipeline to build up the request. In preview 5 we can look into making this implementation more flexible and efficient.

Reply:
Could add a mutable policy that changes the URL of the request based on a `Context` value. That way there would be a single client and single policy to handle the URL switching.